src/coreclr/jit/lower.cpp

```diff
@@ -1613,15 +1613,18 @@ void Lowering::LowerArg(GenTreeCall* call, CallArg* callArg)
         else
         {
             assert(abiInfo.NumSegments == 1);
-            const ABIPassingSegment& stackSeg             = abiInfo.Segment(0);
-            const bool               putInIncomingArgArea = call->IsFastTailCall();
+            // FIXME: Workaround for NumSegments being 0 on Wasm and causing crashes
+            if (abiInfo.NumSegments > 0) {
```

Comment:
When does this happen?

`WasmClassifier::Classify` always returns a single register segment so this seems odd.
Also with the assert above this doesn't make sense.

Reply:
The fix should be to make sure there is some reasonable ABI info there for WASM. But given the implementation of `WasmClassifier` it is odd that there isn't, so we should figure out why.

The assert above will keep firing for people running the JIT in debug/checked so some other fix is needed.

Reply:
That sounds like a good idea. A good place to start would be by obtaining the JITDUMP for the function where the AV occurs. The example at https://github.com/dotnet/runtime/blob/main/docs/workflow/debugging/coreclr/debugging-aot-compilers.md#example-of-debugging-a-test-application-in-crossgen2 might help with that.
Also feel free to hit me up internally.